src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -74,6 +76,34 @@ public RuntimeConfigProvider(RuntimeConfigLoader runtimeConfigLoader)
     /// <exception cref="DataApiBuilderException">Thrown when the loader is unable to load an instance of the config from its known location.</exception>
     public RuntimeConfig GetConfig()
     {
+        // Only used in hot reload to validate the configuration file
+        if (_configLoader.isNewConfigDetected && !_configLoader.isNewConfigValidated)
```

Comment:
your pr description checks the box for "integration tests." how do existing integration tests check hot reload and this new functionality? I'd suggest outline in your PR description how you tested this, even if manually tested.